parser/block.go

```diff
@@ -129,15 +129,18 @@ func (b *Block) ParseFromSlice(data []byte) (rest []byte, err error) {
 	data = []byte(s)
 
 	vtx := make([]*Transaction, 0, txCount)
-	for i := 0; len(data) > 0; i++ {
+	var i int
+	for i = 0; i < txCount && len(data) > 0; i++ {
 		tx := NewTransaction()
 		data, err = tx.ParseFromSlice(data)
 		if err != nil {
 			return nil, errors.Wrap(err, fmt.Sprintf("parsing transaction %d", i))
 		}
 		vtx = append(vtx, tx)
 	}
-
+	if i < txCount {
+		return nil, errors.New("parsing block transactions: not enough data")
+	}
```

Comment:
Previously, if the number of transactions actually in the block was less than the txCount (variable-length integer) in the block, the error would be silently ignored. Note that it's okay for there to be extra data, because it gets returned (below). In general, this parsing code allows sequences of blocks concatenated together, but we don't currently depend on that. The caller should check that no data remains.